statefun-kafka-io/src/test/java/org/apache/flink/statefun/sdk/kafka/KafkaIngressBuilderTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.statefun.sdk.kafka;
+
+import static org.apache.flink.statefun.sdk.kafka.testutils.Matchers.hasProperty;
+import static org.apache.flink.statefun.sdk.kafka.testutils.Matchers.isMapOfSize;
+import static org.apache.flink.statefun.sdk.kafka.testutils.Matchers.startupPositionOf;
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.util.Collections;
+import java.util.Properties;
+import org.apache.flink.statefun.sdk.io.IngressIdentifier;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.junit.Test;
+
+public class KafkaIngressBuilderTest {
+
+  private static final IngressIdentifier<String> DUMMY_ID =
+      new IngressIdentifier<>(String.class, "ns", "name");
+
+  @Test
+  public void exampleUsage() {
```

Comment:
I think that this test can be broken down a little bit, too number of separated tests.
one that validates that the topics are there, and one that valises that the properties are set currently, and one for the serializer.